.github/workflows/lint.yml

```diff
@@ -1,25 +1,41 @@
 name: Lint
+# Lint runs golangci-lint over the entire gaia repository
+# This workflow is run on every pull request and push to main
+# The `golangci` will pass without running if no *.{go, mod, sum} files have been changed.
 on:
   pull_request:
   push:
     branches:
       - main
-
 jobs:
   golangci:
-    name: Run golangci-lint
+    name: golangci-lint
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
       - uses: actions/setup-go@v3
         with:
-          go-version: 1.17
+          go-version: 1.18
       - uses: technote-space/get-diff-action@v6.0.1
+        id: git_diff
         with:
           PATTERNS: |
             **/**.go
             go.mod
             go.sum
-      - name: Run golangci-lint
-        run: make lint
+      - name: Get data from Go build cache
+        # if: env.GIT_DIFF
+        if: ${{ false }}
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/golangci-lint
+            ~/.cache/go-build
+          key: ${{ runner.os }}-go-build-${{ hashFiles('**/go.sum') }}
```

Comment:
```suggestion
```

Reply:
this isn't needed as golangci=lint caches on its own. we need to clean up in the sdk